solutions/숫자정사각형.js

```diff
@@ -0,0 +1,41 @@
+const INPUT_PATH = "../inputs/숫자정사각형.txt";
+
+const fs = require("fs");
+const filePath = process.platform === "linux" ? "/dev/stdin" : INPUT_PATH;
+let [information, ...map] = fs
+  .readFileSync(filePath)
+  .toString()
+  .trim()
+  .split("\n");
+map = map.map((value) => value.split(""));
+
+const [height, width] = information.split(" ").map(Number);
+let max = 0;
+
+for (let i = 0; i < height; i++) {
+  for (let j = 0; j < width; j++) {
+    max = Math.max(max, findSquare(map, i, j, 0, 1) ** 2);
+  }
+}
+
+console.log(max);
+
+function findSquare(map, startX, startY, count, max) {
+  const isOut = startX + count >= width || startY + count >= height;
+
+  if (isOut) {
+    return max;
+  }
+
+  const right = map[startY][startX + count];
+  const down = map[startY + count][startX];
+  const cross = map[startY + count][startX + count];
+
+  if (right === down && down === cross) {
+    max = count;
+  }
+
+  count++;
+
+  return findSquare(map, startX, startY, count, max);
```

Comment:
맥스값 갱신이 조금 잘못된것같습니다.
이경우 max가 1이고 count가 1이면 멕스는 2가되어야하는데 현재는 1로 유지될것같습니다...

```suggestion
  if (right === down && down === cross && map[startY][startX] === right) {
    max = count + 1;
  }

  return findSquare(map, startX, startY, count + 1, max);
```

Reply:
보완해보겠습니다 감사합니당!!

Reply:
다시보니 멕스에 현재값을넣어주는게 맞는것 같습니다..혼란을드려 죄송합니다.